hail/python/hailtop/hailctl/auth/login.py

```diff
@@ -76,21 +76,18 @@ def init_parser(parser):  # pylint: disable=unused-argument
     token = resp['token']
     username = resp['username']
 
-    auth_ns = deploy_config.service_ns('auth')
-    tokens = get_tokens()
-    tokens[auth_ns] = token
     dot_hail_dir = os.path.expanduser('~/.hail')
     if not os.path.exists(dot_hail_dir):
         os.mkdir(dot_hail_dir, mode=0o700)
-    tokens.write()
+    auth_ns = deploy_config.service_ns('auth')
+    Tokens({auth_ns: token}).write()
 
     print(f'Logged in as {username}.')
 
 
 async def async_main():
-    headers = auth_headers('auth', authorize_target=False)
     async with aiohttp.ClientSession(
-            raise_for_status=True, timeout=aiohttp.ClientTimeout(total=60), headers=headers) as session:
+            raise_for_status=True, timeout=aiohttp.ClientTimeout(total=60)) as session:
```

Comment:
You still need to pass the headers.  If you're logging into the `auth` in the dking namespace, you still need to pass the default token to route through `internal.hail.is`.  This is why I pass `authorize_target=False` to the `auth_headers` call: I'm not authorizing the target auth, of course not, because I'm logging in, I'm authorizing the route.